pkg/apis/application/v1alpha1/types.go

```diff
@@ -331,6 +334,14 @@ type SyncPolicy struct {
 type SyncPolicyAutomated struct {
 	// Prune will prune resources automatically as part of automated sync (default: false)
 	Prune bool `json:"prune,omitempty" protobuf:"bytes,1,opt,name=prune"`
+	// SelfHeal applies all app resources repeatedly, regardless of changes to git commit SHA (default: false)
```

Comment:
It looks like it is hard to specify a meaninful value for `maxRetries` field.  E.g. user deploys `nginx` helm chart, don't plan to switch to a new version and just want to make sure that any manual k8s change will be automatically reverted. In this case user wants to retry as much as needed.

On the other hand we want to prevent a situation when Argo CD syncing application every second because of problematic CRD which controller keep changing it.

So instead of specifying `maxRetries` we need to enforce a timeout between retries. In this case `selfHeal` can be just boolean value and timeout can be specified using argocd-application-controller flag.

Reply:
Agreed. Implemented this because of of the comments (wishes 😉 ) in #1736 
Can you elaborate on how the application-controller flag should be implemented, or is already implemented?